### DISCUSSION TITLE
PETSc was configured with MPICH but now appears to be compiling using a non-MPICH mpi.h

### DISCUSSION THREAD

**Discussion by KangChenRui** [idaholab/moose#22145] [Q&A General] [answered]
Hi everyone,
 
yesterday，I used "make",  the report was "PETSc was configured with MPICH but now appears to be compiling using a non-MPICH mpi.h".
 
what should I do ?

Thanks for your help!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You need to be consistent between MPI distributions when building MOOSE and its dependencies. Typically this can happen if you loaded an HPC module when configuring, then loaded another when (re-)building

Or if you have both a conda-installed MPI distribution and a non-conda MPI already on your machine

Guillaume

**Reply by Adairle** [reply]
thanks for your help!
but I don‘t how to operate it.

**Reply by GiudGiud** [reply]
what is the output of the diagnostic script in moose/scripts

**Reply by richman2024** [reply]
I got this problem too, besides another one is "no type named 'value_type' in 'struct std::iterator_traits<infix_ostream_iterator<std::basic_string<char> > >", what shoud I do?

Thank you for your help!

**Reply by GiudGiud** [reply]
Hello

please post in a new discussions thanks
